app/components/InputLabel.tsx

```diff
@@ -1,9 +1,24 @@
 import { PropsWithChildren } from "react";
+import { tv } from "tailwind-variants";
 
-export default function InputLabel({ children }: PropsWithChildren) {
-  return (
-    <label className="font-manrope text-primary-600 font-normal text-xs tracking-wide">
-      {children}
-    </label>
-  );
+type InputLabelProps = {
+  inputState: "error" | "filled" | "empty";
+};
+
+const label = tv({
+  base: "font-manrope text-primary-600 font-normal text-xs tracking-wide",
+  variants: {
+    state: {
+      error: "font-semibold text-complementary-600",
+      filled: "",
+      empty: "",
+    },
+  },
+});
+
+export default function InputLabel({
+  inputState,
+  children,
+}: PropsWithChildren<InputLabelProps>) {
+  return <label className={label({ state: inputState })}>{children}</label>;
```

Comment:
with previous suggestion you could use the [shorthand property](https://ui.dev/shorthand-properties) {label({ state })}